src/components/Quiz.js

```diff
@@ -18,19 +18,56 @@ const Quiz = () => {
     setQuestions(quizData);
   }, [quizName]);
 
-  const handleAnswerSelect = e => {
-    const selectedAnswer = { [e.target.name]: Number(e.target.value) };
+  const handleCheckboxAnswers = (e, answers) => {
+    const questionIds = Object.keys(answers || {});
+    const hasQPreviouslySelectedAnswers =
```

Comment:
As I mentioned in my previous comments, we do not need this variable. Let's remove it. If you are confused why we don't need it then let me know and I'll show you the code that works without it

Reply:
Done!